app/vmalert/utils/auth.go

```diff
@@ -0,0 +1,76 @@
+package utils
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"path/filepath"
+	"strings"
+	"unicode"
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/logger"
+)
+
+// Authorizer is a func which adds Auth header
+// to the given request
+type Authorizer func(r *http.Request)
+
+const authHeader = "Authorization"
+
+// RequestAuthorizer returns Authorizer based on the
+// provided auth params.
+// Returns nil if all auth params are empty.
+func RequestAuthorizer(baUsername, baPassword, bearerToken, bearerTokenFile string) (Authorizer, error) {
+	var fn Authorizer
+	if baPassword != "" {
```

Comment:
As far as i know, its allowed to have basicAuth username with unset/empty password.